sympy/matrices/common.py

```diff
@@ -802,9 +803,8 @@ def entry(i, j):
 
     @classmethod
     def _eval_zeros(cls, rows, cols):
-        def entry(i, j):
-            return cls.zero
-        return cls._new(rows, cols, entry)
+        from sympy.matrices.dense import Matrix
+        return cls._new(Matrix._new(rows, cols, [cls.zero]*(rows*cols), copy=False))
```

Comment:
This shouldn't be importing Matrix and using `Matrix._new`. Maybe the `copy=False` argument should be added somewhere else like in `CommonMatrix`.

Reply:
In the latest commit, I added the `copy` argument in the case it was failing for. This may not be the best fix, but now they are even faster than before.
```
In [1]: %time ok = eye(1000)
Wall time: 3.98 ms

In [2]: %time ok = zeros(1000)
Wall time: 9.97 ms
```